go-controller/pkg/ovn/kubevirt.go

```diff
@@ -110,18 +122,287 @@ func (oc *DefaultNetworkController) deleteDHCPOptions(pod *kapi.Pod) error {
+		return nil
+	}
+
+	podAnnotation, err := util.UnmarshalPodAnnotation(pod.Annotations, types.DefaultNetworkName)
+	if err != nil {
+		return fmt.Errorf("failed reading ovn annotation: %v", err)
+	}
+
+	node, err := oc.watchFactory.GetNode(pod.Spec.NodeName)
+	if err != nil {
+		return err
+	}
+	transitSwitchPortAddrs, err := util.ParseNodeTransitSwitchPortAddrs(node)
+	if err != nil {
+		return err
+	}
+	for _, podIP := range podAnnotation.IPs {
+		ipFamily := utilnet.IPFamilyOfCIDR(podIP)
+		transitSwitchPortAddr, err := util.MatchFirstIPNetFamily(ipFamily == utilnet.IPv6, transitSwitchPortAddrs)
+		if err != nil {
+			return err
+		}
+		route := nbdb.LogicalRouterStaticRoute{
+			IPPrefix: podIP.IP.String(),
+			Nexthop:  transitSwitchPortAddr.IP.String(),
+			Policy:   &nbdb.LogicalRouterStaticRoutePolicyDstIP,
+			ExternalIDs: map[string]string{
+				kubevirt.OvnZoneExternalIDKey:           kubevirt.OvnRemoteZone,
+				string(libovsdbops.VirtualMachineIndex): pod.Labels[kubevirtv1.VirtualMachineNameLabel],
+				string(libovsdbops.NamespaceIndex):      pod.Namespace,
+			},
+		}
+		if err := libovsdbops.CreateOrReplaceLogicalRouterStaticRouteWithPredicate(oc.nbClient, types.OVNClusterRouter, &route, func(item *nbdb.LogicalRouterStaticRoute) bool {
+			matches := item.IPPrefix == route.IPPrefix && item.Nexthop == route.Nexthop && item.Policy != nil && *item.Policy == *route.Policy
+			return matches
+		}); err != nil {
+			return fmt.Errorf("failed adding static route at remote zone: %v", err)
+		}
+	}
+	return nil
+}
+
+func (oc *DefaultNetworkController) ensureRoutingForVM(pod *kapi.Pod) error {
+	isMigratedSourcePodStale, err := kubevirt.IsMigratedSourcePodStale(oc.watchFactory, pod)
+	if err != nil {
+		return err
+	}
+	if util.PodWantsHostNetwork(pod) || !kubevirt.IsPodLiveMigratable(pod) || isMigratedSourcePodStale {
+		return nil
+	}
+
+	targetNode := pod.Labels[kubevirtv1.NodeNameLabel]
+	targetReadyTimestamp := pod.Annotations[kubevirtv1.MigrationTargetReadyTimestamp]
+	// No live migration or target node was reached || qemu is already ready
+	if targetNode == pod.Spec.NodeName || targetReadyTimestamp != "" {
+		if oc.isPodScheduledinLocalZone(pod) {
+			if err := oc.ensureLocalZonePodAddressesToNodeRoute(pod); err != nil {
+				return fmt.Errorf("failed ensureLocalZonePodAddressesToNodeRoute for %s/%s: %w", pod.Namespace, pod.Name, err)
+			}
+		} else {
+			if err := oc.ensureRemoteZonePodAddressesToNodeRoute(pod); err != nil {
+				return fmt.Errorf("failed ensureRemoteZonePodAddressesToNodeRoute for %s/%s: %w", pod.Namespace, pod.Name, err)
 			}
 		}
 	}
 	return nil
 }
+
+func (oc *DefaultNetworkController) syncVirtualMachines(vms map[ktypes.NamespacedName]bool) error {
+	// If there is no ovn_cluster_router OVN garbage collector will do the job
+	_, err := libovsdbops.GetLogicalRouter(oc.nbClient, &nbdb.LogicalRouter{Name: ovntypes.OVNClusterRouter})
+	if err == libovsdbclient.ErrNotFound {
+		return nil
+	}
```

Comment:
You don't need to do this anymore. We changed it so that the delete ops wont fail if if the router doesn't  exist.